Fix/add tooltip issue 3682

mkbeefcake opened the pull request:
Fixed the issue https://github.com/Joystream/pioneer/issues/3682

traumschule commented on packages/ui/src/council/components/election/CandidateVote/CandidateVote.tsx:
```diff
@@ -76,10 +77,17 @@ export const CandidateVote = ({
                 <StatsValue>
                   <TokenValue value={ownStake} />
                 </StatsValue>
+                <ReactTooltip id="tooltipForVotes" multiline={true} />
```
You can use 
```
import { Tooltip, TooltipDefault } from '@/common/components/Tooltip'

         <Tooltip tooltipTitle="" tooltipText="" tooltipLinkURL="">
            <TooltipDefault />
          </Tooltip>
```

mkbeefcake replied:
Thanks for your suggestion.
I followed you and updated the code.

traumschule replied:
The voting stake part looks good now. I think we can drop the `TooltipDefault` because it already has a child.

mkbeefcake replied:
I removed that line

traumschule commented on packages/ui/src/council/components/election/CandidateVote/CandidateVote.tsx:
```diff
@@ -76,7 +79,9 @@ export const CandidateVote = ({
             {hasOwnStake && (
               <>
-                <Subscription>My Stake</Subscription>
+                <Tooltip tooltipText={tooltipLabel}>
```
add `tooltipLinkURL="https://joystream.gitbook.io/testnet-workspace/system/council#candidacy"`

mkbeefcake replied:
yes i will add it

traumschule commented on packages/ui/src/council/components/election/CandidateVote/CandidateVote.tsx:
```diff
@@ -76,7 +79,9 @@ export const CandidateVote = ({
               <>
-                <Subscription>My Stake</Subscription>
+                <Tooltip tooltipText={tooltipLabel}>
+                  <Subscription>My contributed votes</Subscription>
```
Just to confirm, is this being used on both Reveal Vote screen and Past Elections?

mkbeefcake replied:
> Just to confirm, is this being used on both Reveal Vote screen and Past Elections?

yes. I guess so. I confirmed it on Past Elections screen in my end, and but couldn't check it "Reveal Vote" screen, because i couldn't be attended on Vote screen. but in the code level, I checked those 2 screens are using same components.

traumschule replied:
ok, we can leave that to QA.

traumschule reviewed (changes requested):
Wasn't sure if this is still in progress but it looks like, sorry for the confusion.
- [X] Add Candidacy staking tooltip
- [X] On both! Reveal Vote screen and Past Elections for "My stake"
- [X] Rename to "My contributed votes"
- [ ] Add Voting stake tooltip with link

We also need to fix the tests.
- [X] In `packages/ui/test/council/pages/PastElection.test.tsx` replace line 177 (was "My stake") with
```
expect(queryAllByText(/My contributed votes/i).length).toBe(1)
```
- [X] In `packages/ui/test/council/components/ElectionVotes.test.tsx` replace line 123 (was "my stake") with 
```
expect((await screen.findByText(/My contributed votes/)).nextSibling?.textContent).toEqual('2,000')
```

mkbeefcake commented:
> Wasn't sure if this is still in progress but it looks like, sorry for the confusion.
> 
> * [ ]  Add Candidacy staking tooltip => add link
> * [ ]  On both! Reveal Vote screen and Past Elections for "My stake" <= please confirm below
> * [x]  Rename to "My contributed votes"
> * [ ]  Add Voting stake tooltip with link
> 
> We also need to fix the tests.
> 
> * [ ]  In `packages/ui/test/council/pages/PastElection.test.tsx` replace line 177 (was "My stake") with
> 
> ```
> expect(queryAllByText(/My contributed votes/i).length).toBe(1)
> ```
> 
> * [ ]  In `packages/ui/test/council/components/ElectionVotes.test.tsx` replace line 123 (was "my stake") with
> 
> ```
> expect((await screen.findByText(/My contributed votes/)).nextSibling?.textContent).toEqual('2,000')
> ```

I updated the test code and added tooltiplink, please check it in your end

traumschule commented:
```
 | FAIL test/council/components/ElectionVotes.test.tsx (7.546 s)
 |   ● UI: RevealingStageVotes › Own stake
 |
 |     expect(received).toEqual(expected) // deep equality
 |
 |     Expected: "2,000"
 |     Received: undefined
 |
 |       122 |     expect((await screen.findByText(/total stake/i)).nextSibling?.textContent).toEqual('3,000')
 |     > 123 |     expect((await screen.findByText(/My contributed votes/i)).nextSibling?.textContent).toEqual('2,000')
```

oleksanderkorn reviewed (changes requested):
Check failing tests.

traumschule reviewed (changes requested):
Please resolve the conflicts.

mkbeefcake commented:
> Please resolve the conflicts.

I solved the conflicts. please check and give me feedback
